modules/vaos/app/models/vaos/message_form.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'active_model'
+require 'common/models/form'
+
+module VAOS
+  class MessageForm < Common::Form
+    attribute :message_text, String # only this attribute gets set by user, everything else is overridden
+    attribute :url, String
+    attribute :sender_id, String
+    attribute :appointment_request_id, String
+    attribute :message_date_time, String
+    attribute :message_sent, Boolean
+    attribute :is_last_message, Boolean
+    attribute :_appointment_request_id, String
+
+    def initialize(user, request_id, json_hash = {})
+      super(json_hash)
+      @user = user
+      @appointment_request_id = request_id
+    end
+
+    # HACK: addresses the fact that downstream requires this as AppointmentRequestId (upper camelcase)
```

Comment:
both `appointment_require_id` and `_appointment_request_id` become `AppointmentRequestId` when using `.camelize` -- this comment really doesn't make it clear where or why you need the extra variable

Reply:
NOT TRUE. Camelize as it's written in the middleware does lowerCamelCase. All other attributes in the request are in fact required to be lowerCamelCase, but this particular attribute for whatever reason must be UpperCamelCase.

Reply:
Can we add a test to definitively show this?

Reply:
Maybe just take of the word hack? To me, "hack" implies that it could/should be done better and that it should be looked at later.

Reply:
OK, that totally makes sense to me. I've changed it to:

```ruby
# DOWNSTREAM_BUG: addresses the fact that downstream requires this as AppointmentRequestId (upper camelcase)
```